lib/domain/usecase/implementation/get_movie_genres_usecase.dart

```diff
@@ -1,15 +1,44 @@
+import 'package:connectivity_plus/connectivity_plus.dart';
+
 import '../../../data/model/data_state.dart';
 
 import '../../../data/repository/genre_repository.dart';
 import '../../entity/genre.dart';
+import '../../repository/genre_database_repository_interface.dart';
 import '../../repository/genre_repository_interface.dart';
 import '../usecase_interface.dart';
 
 class GetMovieGenresUseCase implements UseCase<DataState> {
   IGenreRepository genreRepository = GenreRepositoryImp();
+  IGenreDatabaseRepository genreDatabaseRepository;
+
+  GetMovieGenresUseCase({
+    required this.genreDatabaseRepository,
+  });
 
   @override
-  Future<DataState<List<Genre>>> call({params}) {
-    return genreRepository.getMovieGenres(params);
+  Future<DataState<List<Genre>>> call({params}) async {
+    var connectivityResult = await (Connectivity().checkConnectivity());
+    if (connectivityResult == ConnectivityResult.none) {
+      try {
+        List<Genre> genres =
+            await genreDatabaseRepository.getMovieGenres(params);
+        return DataSuccess(genres);
+      } on Exception catch (e) {
+        return DataFailed(e.toString());
```

Comment:
Why do you need to have the try-catch block here

Reply:
done